require that `TARGET_DET_LIST` is sorted (and sort it here)

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63400
* __->__ #64102

We sort this list so that we may add comments to indicate the absence
of a file right where that file would need to be put. This makes it
difficult to wrongly add such a file.

The sorting itself was done programmatically to ensure that no entries
were inadvertently removed.

I printed the sorted list with:

```
  for p in sorted(TARGET_DET_LIST):
    print(f'    "{p}",')
```

Then copied it back into the file.

Differential Revision: [D30625076](https://our.internmc.facebook.com/intern/diff/D30625076)

Commit:
require that TARGET_DET_LIST is sorted (and sort it here)

We sort this list so that we may add comments to indicate the absence
of a file right where that file would need to be put. This makes it
difficult to wrongly add such a file.

The sorting itself was done programmatically to ensure that no entries
were inadvertently removed.

I printed the sorted list with:

```
  for p in sorted(TARGET_DET_LIST):
    print(f'    "{p}",')
```

Then copied it back into the file.

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30625076).

Commit:
Update on "require that `TARGET_DET_LIST` is sorted (and sort it here)"


We sort this list so that we may add comments to indicate the absence
of a file right where that file would need to be put. This makes it
difficult to wrongly add such a file.

The sorting itself was done programmatically to ensure that no entries
were inadvertently removed.

I printed the sorted list with:

```
  for p in sorted(TARGET_DET_LIST):
    print(f'    "{p}",')
```

Then copied it back into the file.

Differential Revision: [D30625076](https://our.internmc.facebook.com/intern/diff/D30625076)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30625076).

albanD commented on tools/testing/modulefinder_determinator.py:
```diff
@@ -78,8 +37,64 @@
 
+
+def assert_is_sorted(seq: Sequence[str], name: str) -> None:
+    diff = '\n'.join(difflib.unified_diff(seq, sorted(seq)))
```
Why do you need difflib here?

dagitses replied:
If it's wrong, this shows you where it's not sorted.

albanD reviewed (commented):
I am not 100% sure to follow why we need that.

dagitses commented:
> I am not 100% sure to follow why we need that.

if we don't, it's hard to robustly prevent someone from adding a specific file that should not be there. They could put it at the top or the bottom, or anywhere else.

If it's sorted, there's exactly one place it can go and you can leave a comment there saying don't add that entry.

albanD commented:
Ok. I'll let @driazati judge what to do here.

dagitses commented on tools/testing/modulefinder_determinator.py:
```diff
@@ -78,8 +37,64 @@
+        raise AssertionError(f'{name} is not sorted:\n{diff}')
+
+
+# We keep TARGET_DET_LIST sorted to minimize merge conflicts but most
+# importantly to allow us to comment on the absence of a test. It
+# would be very difficult to add a file right next to a comment that
+# says to keep it out of the list.
+assert_is_sorted(TARGET_DET_LIST, 'TARGET_DET_LIST')
```
Not a VSCode user, but good to know.

I moved the assertion to test_determination.py. What do you think about that?

Commit:
Update on "require that `TARGET_DET_LIST` is sorted (and sort it here)"


We sort this list so that we may add comments to indicate the absence
of a file right where that file would need to be put. This makes it
difficult to wrongly add such a file.

The sorting itself was done programmatically to ensure that no entries
were inadvertently removed.

I printed the sorted list with:

```
  for p in sorted(TARGET_DET_LIST):
    print(f'    "{p}",')
```

Then copied it back into the file.

Differential Revision: [D30625076](https://our.internmc.facebook.com/intern/diff/D30625076)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30625076).
